### DISCUSSION TITLE
Reading a variable, external amount of data into a struct at runtime

### DISCUSSION THREAD

**Discussion by pbar1** [jam1garner/binrw#261] [Q&A]
Let's say I've got a struct that looks like this:

```rust
#[binrw]
pub struct MyPacket {
    #[br(/* this can vary at runtime */)]
    #[bw(try_calc(u32::try_from(payload.len())))]
    pub length: u32,

    #[br(count = length)]
    pub payload: Vec<u8>,
}
```

**How would I go about setting `length` for the purposes of `BinRead`** if it can vary at runtime according to external conditions (say the program's state) rather than in relation to any of the other fields of `MyPacket`?

I can't make use of something like `until_eof` helper because this is part of a stream.

**Comment by kitlith** [comment]
You can add arguments to the struct using `#[br(import(...))` and similar, to be given using `#[br(args(...)]` or `BinRead::read_options(...)`
